contracts/SoulBoundToken/BasicSBT.sol

```diff
@@ -0,0 +1,215 @@
+// SPDX-License-Identifier: MIT
+// SpartanLabs Contracts (SBT)
+pragma solidity ^0.8.0;
+
+import "./utils/Ownable.sol";
+
+/**
+ * @dev Implementation of Soul Bound Token (SBT)
+ * following Vitalik's co-authored whitepaper at: https://papers.ssrn.com/sol3/papers.cfm?abstract_id=4105763
+ *
+ * Contract provides a basic Soul Bound Token mechanism, where address can mint SBT with their private data.
+ *
+ * By default, the owner account will be the one that deploys the contract. This
+ * can later be changed with {transferOwnership}.
+ */
+contract BasicSBT is Ownable {
+    // Name for the SBT
+    string public _name;
+
+    // Symbol for the SBT
+    string public _symbol;
+
+    // Total count of SBT
+    uint256 public _totalSBT; // TODO: Should we keep track of how many SBT?
+
+    // Mapping from SBT ID to owner address
+    mapping(uint256 => address) private _owners; // TODO : Do we even need to keep track of owners?
+
+    // Mapping between address and the soul
+    mapping(address => Soul) private souls;
```

Comment:
Considering that `BasicSBT` is similar to `BasicNFT` - it would be good to follow https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/ERC721.sol from L30 to L39 - that has been thought through thoroughly based on ERC721

Reply:
I think the main difference is whether we want to index those tokens. For NFTs, it makes sense because we want a way to identify these tokens and it makes generation of NFT art based on layer and tagging it to a token id easy. 

However for SBT I am not sure how token id might be needed?

Reply:
for these lines
```
    // Mapping from token ID to approved address
    mapping(uint256 => address) private _tokenApprovals;

    // Mapping from owner to operator approvals
    mapping(address => mapping(address => bool)) private _operatorApprovals;
    ```
    
    we do not have approvals, so we may not need to include them. 

ERC721 has approval because we need to approve protocols transfer.

initially was thinking of approval for editing or viewing, but does not make much sense that we need approval to view.

However, we could have approval for edit for different protocols? Maybe that is something we can consider as different protocols might want to edit the SBT. But philosophically, I think only the owner of the project should be the one editing SBT. 

What do you think?

Reply:
Yeap the owner should be the one editing, approval to view for basic is not necessary but if we go down the more privacy centric paths we need it right?

Reply:
`However for SBT I am not sure how token id might be needed?` yeah don't see why it's needed also - ok understood

Reply:
> Yeap the owner should be the one editing, approval to view for basic is not necessary but if we go down the more privacy centric paths we need it right?

i think that if we were to go down privacy centric path, 

- if the data is encrypted we do not need it
- if the data is not encrypted, we need approvals. however as you mentioned, malicious attackers can look into the block space to find the decrypted data held within the mapping

Therefore I realised that approvals does not do anything to protect the privacy of the SBT holders

Reply:
so i guess 

- approvals is not needed
- token id is not needed

will remove the line
`mapping(uint256 => address) private _owners;`

Reply:
👌